tests/query/atomWithInfiniteQuery.test.tsx

```diff
@@ -216,76 +216,3 @@ it('infinite query with enabled 2', async () => {
   fireEvent.click(getByText('set enabled'))
   await findByText('slug: hello-world')
 })
-
-// adapted from https://github.com/tannerlinsley/react-query/commit/f9b23fcae9c5d45e3985df4519dd8f78a9fa364e#diff-121ad879f17e2b996ac2c01b4250996c79ffdb6b7efcb5f1ddf719ac00546d14R597
-it('should be able to refetch only specific pages when refetchPages is provided', async () => {
```

Comment:
Why can't we keep this test?

Reply:
> Why can't we keep this test?

https://github.com/pmndrs/jotai/pull/1126#discussion_r861380333
u should see this reply u propose😂, so i remove the test of `refetch`.
but oddly i found that `useInfiniteQuery` still returns `refetch` method after my check just now
so we should still keep this?

Reply:
😿 Yeah, sorry, I change my mind...
I start to think about #1199.
So, this PR should be minimal to move forward.
I feel like we need some kind of `refetch`, which can be different from what we have now, but it's out of the scope of this PR anyway.